evm/vm/forks/sharding/collations.py

```diff
@@ -26,6 +26,8 @@ def __init__(self, header, transactions=None, witness_nodes=None):
         if witness_nodes is None:
             witness_nodes = []
 
+        self.transaction_fee_sum = 0
```

Comment:
This doesn't appear to be used (as researched by a quick ctrl+f)

Reply:
`block.transaction_fee_sum`(or `collation.transaction_fee_sum`) is used to record the accumulated transaction fee paid by each transaction that's included in the block(collation).

```python
def add_transaction(self, transaction, computation, block):
    …
    # Calculate transaction fee
    transaction_fee, _ = computation.compute_transaction_fee_and_refund()
    # Bookkeep this transaction fee
    block.transaction_fee_sum += transaction_fee
    ...
```

Reply:
Ah, I see.

I've opened #412 to track the task of trying to find a new way to do this.

I need to dig in and understand a bit better why this is there in the first place, but I suspect we can find a different mechanism.